helper-cli/src/main/kotlin/commands/ConvertToDependencyGraphCommand.kt

```
@@ -0,0 +1,153 @@
+
+    override fun run() {
+        if (!convertOrtResult() && !convertAnalyzerResult() && !convertProjectAnalyzerResult()) {
+            failure("No supported data type could be found in the input file.", 1)
+        }
+    }
+
+    /**
+     * Convert a full ORT result file.
+     */
+    private fun convertOrtResult(): Boolean = convertResult<OrtResult> { result ->
+        println("Converting a full ORT result file.")
+
+        val analyzerResult = convertToGraph(result.analyzer?.result)
+        val convertedResult = result.copy(
+            analyzer = result.analyzer?.copy(result = analyzerResult)
+        )
+
+        writeResult(convertedResult)
+    }
+
+    /**
+     * Convert an [AnalyzerResult] file.
+     */
+    private fun convertAnalyzerResult(): Boolean = convertResult<AnalyzerResult> {
+        println("Converting an analyzer result file.")
+
+        writeResult(convertToGraph(it))
+    }
+
+    /**
+     * Convert a [ProjectAnalyzerResult]. In this case, the result file contains an [AnalyzerResult], as a
+     * [ProjectAnalyzerResult] does not contain a dependency graph.
+     */
+    private fun convertProjectAnalyzerResult(): Boolean = convertResult<ProjectAnalyzerResult> { result ->
+        println("Converting a project analyzer result file.")
+
+        val analyzerResult = AnalyzerResultBuilder().addResult(result).build()
+        writeResult(convertToGraph(analyzerResult))
+    }
+
+    /**
+     * Convert a result of type [T]. Read this result from the input file and invoke [block] with it. Return *false*
+     * if the result file could not be loaded (which probably means that it contains a result of a different type).
+     */
+    private inline fun <reified T : Any> convertResult(block: (T) -> Unit): Boolean =
+        try {
+            block(ortFile.readValue())
+            true
+        } catch (e: JsonMappingException) {
+            println("Could not convert result type ${T::class.simpleName}: ${e.message}")
```

Isn't it a bit confusing to see this message while "auto-detecting" the input file format? IMO we could drop this, and only keep the error message if no supported input file format could be detected.

---

The main reason for this message was `detekt` complaining about a swallowed exception. This no longer applies due to the `runCatching` construct. So the message has been dropped.